src/lib/structures/uid.mli

```diff
@@ -30,10 +30,22 @@ module DE = Dolmen.Std.Expr
 type t = private
   | Hstring : Hstring.t -> t
   | Dolmen : 'a DE.id -> t
+  | Term_cst : DE.term_cst -> t
+  | Ty_cst : DE.ty_cst -> t
 
 val of_dolmen : 'a DE.Id.t -> t
+val of_term_cst : DE.term_cst -> t
+val of_ty_cst : DE.ty_cst -> t
 val of_string : string -> t
 val of_hstring : Hstring.t -> t
+
+val to_term_cst : t -> DE.term_cst
+
+val order_tag : int Dolmen.Std.Tag.t
+(** Tag used to attach the order of constructor. Used to
+    retrieve efficiency the order of the constructor in [to_int]. *)
```

Comment:
Nit: in the current design, I believe moving `order_tag` and `perfect_hash` to `Nest` would allow hiding `order_tag` from the interface.